src/Form/FormAnalyzer.js

```diff
@@ -229,6 +232,92 @@ class FormAnalyzer {
         });
     }
 
+    /**
+     * Takes an element and returns all its children that are text-only nodes
+     * @param {HTMLElement|Element} element
+     * @param {number} maxDepth
+     * @param {number} currentDepth
+     * @returns {HTMLElement[]|Element[]}
+     */
+    getElementsWithOnlyTextChild(element, maxDepth = 2, currentDepth = 0) {
+        // Array to collect elements with only text child nodes
+        const elementsWithTextChild = [];
+
+        // If we've reached the max depth, stop traversing further
+        if (currentDepth > maxDepth) {
+            return elementsWithTextChild;
+        }
+
+        // Check if the current element has only one text child node
+        if (element.nodeType === Node.ELEMENT_NODE) {
+            const childNodes = element.childNodes;
+
+            if (childNodes.length === 1 && childNodes[0].nodeType === Node.TEXT_NODE) {
+                elementsWithTextChild.push(element);
+            }
+        }
+
+        // Recurse through each child element and collect matching elements
+        for (const child of element.children) {
+            // Recursively get elements from child elements, increasing depth by 1
+            elementsWithTextChild.push(...this.getElementsWithOnlyTextChild(child, maxDepth, currentDepth + 1));
+        }
+
+        return elementsWithTextChild;
+    }
+
+    evaluateFormHeaderSignals() {
```

Comment:
I don't feel comfortable with this. It's a specific carveout for one site, but it's rather complex. It only gets children of the form parent, scans 2-levels deep, and relies on element size (meaning the form must be visible at the time of the scan), thus reducing the effectiveness massively for the wider web. On the other hand, if we increase the scope it becomes dangerously expensive. I don't know, it seems too specific to warrant such complexity. This form might be something we address with the future remote config project.

I feel like we could/should improve how we use headings, but I'm not sure this is the right way.